### PR TITLE
TMBuilder v3.2.0 - support explicit null checks for nodes

### PR DESCRIPTION
**Pull request by yarg0007**
Request was made to define checks that verify nodes are null. v3.2.0 adds this support. Includes changes to JPChecks*, TMJPCheck* wrappers and all related unit tests.

### CODE REVIEW

File: NST/src/main/java/com/ebay/jsonpath/JPIntegerCheck.java [hunk 1]
```diff
@@ -59,14 +71,18 @@ public void processJsonPath(String jsonPath, SoftAssert softAssert, DocumentCont
       return;
     }
 
-    softAssert.assertNotNull(value, AssertMessageBuilder.build(jsonPath, "with null integer."));
+    if (isNull) {
```
Comment (juliandesmarais) [inline NST/src/main/java/com/ebay/jsonpath/JPIntegerCheck.java:74]: Shouldn't this be isNullExpected ? @byarger-ebay

Comment (yarg0007) [reply NST/src/main/java/com/ebay/jsonpath/JPIntegerCheck.java]: Yes, good catch. Thanks!

File: NST/src/main/java/com/ebay/jsonpath/JPListOfIntegerCheck.java [hunk 2]
```diff
@@ -190,7 +203,9 @@ public void processJsonPath(String jsonPath, SoftAssert softAssert, DocumentCont
       return;
     }
 
-    softAssert.assertNotNull(values, AssertMessageBuilder.build(jsonPath, "because the path does not exist."));
+    if (!isNullExpected()) {
```
Comment (shreyasmp) [inline NST/src/main/java/com/ebay/jsonpath/JPListOfIntegerCheck.java:206]: For collection types, is it worth adding another check empty as well for size ?

**Review by shreyasmp — APPROVED** [review]
Looks fine so far.

File: NST/src/main/java/com/ebay/jsonpath/JPBooleanCheck.java [hunk 3]
```diff
@@ -7,14 +7,15 @@
...
-public class JPBooleanCheck implements JsonPathExecutor, Serializable {
+public class JPBooleanCheck implements JsonPathExecutor, NullCheck<JPBooleanCheck>, Serializable {
 
   /**
    * Serialized ID.
    */
   private static final long serialVersionUID = 888395578771081432L;
 
   private Boolean expectedValue = null;
+  private boolean isNull = false;
```
Comment (juliandesmarais) [inline NST/src/main/java/com/ebay/jsonpath/JPBooleanCheck.java:18]:
```suggestion
  private boolean nullExpected = false;
```

**Review by juliandesmarais — APPROVED** [review]
1 suggestion to rename a var

File: NST/src/main/java/com/ebay/jsonpath/JPListOfIntegerCheck.java (hunk 2 above)
Comment (yarg0007) [reply NST/src/main/java/com/ebay/jsonpath/JPListOfIntegerCheck.java:206]: Empty and size can both be evaluated by using the expected size check that already exists: https://github.com/eBay/NSTSuite/blob/main/NST/src/main/java/com/ebay/jsonpath/JPListOfIntegerCheck.java#L19

File: NST/src/main/java/com/ebay/jsonpath/JPBooleanCheck.java (hunk 3 above)
Comment (yarg0007) [reply NST/src/main/java/com/ebay/jsonpath/JPBooleanCheck.java]: Fixed.

**Pull request merged**